bootstrap_salt/fab_tasks.py

```diff
@@ -28,32 +30,15 @@
 def aws(x):
     env.aws = str(x).lower()
 
+
 @task
 def setup(stack_name=None):
-    if stack_name:
-        env.stack = stack_name
-
-    if env.stack is None:
-        if env.application is None or env.environment is None:
-            print "\n[ERROR] Please specify a stack_id, e.g 'stack:<stack_id>'"
-            sys.exit(1)
-        else:
-            env.stack = "{0}-{1}".format(env.application, env.environment)
-
-    # Setup
+    _validate_fabric_env()
+    env.stack = get_stack_name()
```

Comment:
So my only concern here is that `get_stack_name` looks at `env.stack_name` (and if not found returns a default) and we then store that into env.stack.

- Why have two env vars? Can we not just assign to env.stack_name here?
- Why do we assign to `env.stack = stack_name` at all?

But given this is currently blocking everyone it's probably best to merge this for now and tidy up later.